pkg/channel/channel.go

```diff
@@ -78,3 +79,33 @@ func ListChannel(osnURL string, caCertPool *x509.CertPool, tlsClientCert tls.Cer
 
 	return channels, nil
 }
+
+func ListChannelOnPeer(connection grpc.ClientConnInterface, id identity.SigningIdentity) ([]*peer.ChannelInfo, error) {
```

Comment:
The service invocation is still being done using `context.Background()`, which gives the caller no control over timeout or cancellation of the network operation. Similar to all the function in the chaincode package, I think this should include a caller-supplied Content in the parameters:
```suggestion
func ListChannelOnPeer(ctx context.Content, connection grpc.ClientConnInterface, id identity.SigningIdentity) ([]*peer.ChannelInfo, error) {
```